zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnUserTaskBehavior.java

```diff
@@ -323,10 +333,41 @@ public Optional<UserTaskRecord> userTaskCanceling(final ElementInstance elementI
       return Optional.empty();
     }
 
+    rejectOngoingRequestsForUserTaskBeforeCancellation(userTask);
+
     stateWriter.appendFollowUpEvent(userTaskKey, UserTaskIntent.CANCELING, userTask);
     return Optional.of(userTask);
   }
 
+  private void rejectOngoingRequestsForUserTaskBeforeCancellation(final UserTaskRecord userTask) {
+    asyncRequestState
+        .findAllRequestsByScopeKey(userTask.getElementInstanceKey())
+        .forEach(
+            request -> {
+              switch (request.valueType()) {
+                case USER_TASK ->
+                    responseWriter.writeRejection(
+                        userTask.getUserTaskKey(),
+                        request.intent(),
+                        userTask,
+                        ValueType.USER_TASK,
+                        RejectionType.INVALID_STATE,
+                        "user task was canceled during ongoing transition",
```

Comment:
🔧 I'm wondering whether wording the message around the requests makes more sense. Feel free to take, adjust, or ignore.

```suggestion
                        "user task was canceled before request was finished",
```